jeison.el

```diff
@@ -148,30 +237,50 @@ JSON is an `alist' representing a JSON object where we want information
 to be parsed from.
 PATH is a `list' of keys we should consequently find in JSON and
 proceed with a nested JSON further on."
-  (let* ((json (jeison--read-path json path))
+  (let* ((json-content (jeison--read-path json path))
          (result
           (pcase type
             ;; type is a jeison class: read JSON as a class
-            ((pred jeison-class-p) (jeison--read-class type json))
+            ((pred jeison-class-p) (jeison--read-class type json-content))
             ;; type is a homogeneous list of some sort
-            (`(list-of ,element-type)
-             ;; first we should check that JSON object itself is
-             ;; some sort of sequence
-             (or (cl-typep json 'sequence)
-                 (signal 'jeison-wrong-parsed-type
-                         (list 'sequence json)))
-             ;; then we should iterate of this JSON sequence
-             ;; and parse element-type (retrieved from the type)
-             ;; from each element
-             (mapcar (lambda (element)
-                       (jeison--read-internal element-type element)) json))
```

Comment:
The `mapcar` is your original code. :-)

The `let` just assigns the resulting list to `read-data`, which either gets returned unmolested in the `list-of` case or converted to `vector` in the `jeison-vector-of` case. Perhaps it could be shifted; I wanted to reduce the amount of duplicated code.

Reply:
This comment is obviously misplaced, but my point still stands. In the `cond` condition, you check for `(or IS-LIST IS-VECTOR)` and then at the end of the case, you check which one of them was it. I do understand that it is for reducing duplication and I suggest to extract a common part into a separate function `jeison--read-sequence` (or `jeison--read-array` to be closer to JSON terminology). Then condition will look very simple:
```
(`(list-of ,element-type) (jeison--read-sequence ARGS))
(`(jeison-vector-of ,element-type) (vector (jeison--read-sequence ARGS))
```
I believe that this way is cleaner.

Reply:
@bscottm and this?

Reply:
???